### PR TITLE
Jenkins Git-Magic scripts now sync the indicated branch when doing a Gated Checkin

### PR DESCRIPTION
**Pull request by pgilmorepf**
This was initially a test PR for testing that the changes to Git magic scripts would support gated checkins.
Now, this PR "is" those changes, as well as the proof that they work.

### CODE REVIEW

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
run tests

**Comment by pgilmorepf** [comment]
run tests

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

File: JenkinsConsoleUtility/JenkinsScripts/testInit.sh [hunk 1]
```diff
@@ -11,75 +11,45 @@ CheckVerticalizedParameters
...
-    # cmd <<< "nuget restore Server.sln"
-    # popd
+}
+
+# USAGE: ResetRepo
+ResetRepo (){
+    echo === ResetRepo $PWD, $@ ===
+
+    # Assumes the current directory is set to the repo to be reset
+    SetGitHubCreds
```
Comment (toddbellMSFT) [inline JenkinsConsoleUtility/JenkinsScripts/testInit.sh]: I see this call was here before, but I don't see a function declaration. What is this line in particular doing?

File: JenkinsConsoleUtility/JenkinsScripts/testInit.sh [hunk 2]
```diff
@@ -11,75 +11,45 @@ CheckVerticalizedParameters
...
+    git fetch --progress origin
+    git checkout master || git checkout -b master || CleanCurrentRepo
+    git pull origin master
+
+    if [ "$GitDestBranch"!="master" ]; then
+        git fetch --progress origin
+        git branch -D $GitDestBranch || true
```
Comment (MichaelMackendMS) [inline JenkinsConsoleUtility/JenkinsScripts/testInit.sh]: What does "|| true" actually accomplish here?

**Comment by toddbellMSFT** [comment]
retest

**Pull request merged**